### PR TITLE
feat(healthz): add version

### PR DESCRIPTION
**Pull request by FrankYang0529**
resolve https://github.com/deislabs/bindle/issues/350
ref https://github.com/fermyon/spin/pull/786#issuecomment-1283146177

**Test step**
1. Build.
```
> make build
```
2. Run the server.
```
> ./target/debug/bindle-server --unauthenticated
```
3. Test `/healthz`.
```
> curl -i http://localhost:8080/healthz
HTTP/1.1 200 OK
content-type: application/json
content-length: 38
date: Thu, 27 Oct 2022 11:37:31 GMT

{"status":"OK","version":"0.9.0-rc.1"}
```

### CODE REVIEW

**Review by vdice — COMMENTED** [review]
Nice, this looks good to me.  Ping @thomastaylor312 for a second review.

File: Cargo.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ maintenance = { status = "actively-developed" }
 
 [features]
 default = ["server", "client", "caching", "test-tools", "native-tls"]
-server = ["warp", "openid", "hyper", "mime", "either", "_common"]
+server = ["clap", "warp", "openid", "hyper", "mime", "either", "_common"]
```
Comment (thomastaylor312) [inline Cargo.toml:30]: We should not be bringing in `clap` for a server dependency. It is fairly big and we don't want people to import it unless they are actually building the CLI.

File: src/server/routes.rs [hunk 2]
```diff
@@ -22,7 +29,12 @@ where
     Authn: crate::authn::Authenticator + Clone + Send + Sync + 'static,
     Authz: crate::authz::Authorizer + Clone + Send + Sync + 'static,
 {
-    let health = warp::path("healthz").map(|| "OK");
+    let health = warp::path("healthz").map(|| {
+        warp::reply::json(&HealthStatus {
+            status: "OK".to_string(),
+            version: clap::crate_version!().to_string(),
```
Comment (thomastaylor312) [inline src/server/routes.rs:35]: This can be swapped out to a non-clap thing by doing `env!("CARGO_PKG_VERSION")`

File: src/server/routes.rs [hunk 3]
```diff
@@ -1,9 +1,16 @@
 use std::sync::Arc;
 
+use serde::{Deserialize, Serialize};
 use warp::Filter;
 
 use crate::{server::filters, signature::KeyRing};
 
+#[derive(Deserialize, Serialize)]
+pub struct HealthStatus {
```
Comment (thomastaylor312) [inline src/server/routes.rs:9]: Is this a type that we expect people to read back from the server using a client? If so, would it be better to put into `invoice/api.rs` or another location?

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (FrankYang0529) [reply Cargo.toml]: Removed it. Thanks!

File: src/server/routes.rs (hunk 2 above)
Comment (FrankYang0529) [reply src/server/routes.rs]: Updated it. Thank you.

File: src/server/routes.rs (hunk 3 above)
Comment (FrankYang0529) [reply src/server/routes.rs]: Moved it to `invoice/api.rs` and also changed the name to `HealthResponse`.

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**